apisix/plugin.lua

```diff
@@ -415,6 +415,12 @@ local function meta_filter(ctx, plugin_name, plugin_conf)
         return true
     end
 
+    local match_cache_key =
+        ctx.conf_type .. "#" .. plugin_name .. "#meta_filter_matched"
```

Comment:
Does this key work with multiple global_rules? They have the same conf_type.

Reply:
Do you mean the scene where the `con_type` is the same and the `plugin_name` is different? I think my test case already contains. The `conf_type` is `route`.

Reply:
@soulbird 
Multiple global rules will share the same conf_type
https://github.com/apache/apisix/blob/e2d0f0b16bb157412684a38b129d12acb4628d39/apisix/plugin.lua#L1003

Will this affect the cache of eval result?

Reply:
There is indeed a problem, I adjusted `match_cache_key`, added some test case：https://github.com/apache/apisix/pull/8162/files#diff-29e2373df60fd8fc082317d17e636b6fa17c60459c1a691c5838f36b31cc486aR532-R567